sdks/java/io/kinesis/src/test/java/org/apache/beam/sdk/io/kinesis/KinesisIOIT.java

```diff
@@ -35,33 +38,45 @@
 import org.apache.beam.sdk.values.PCollection;
 import org.joda.time.Duration;
 import org.joda.time.Instant;
+import org.junit.AfterClass;
 import org.junit.BeforeClass;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
+import org.testcontainers.containers.localstack.LocalStackContainer;
 
 /**
  * Integration test, that writes and reads data to and from real Kinesis. You need to provide {@link
- * KinesisTestOptions} in order to run this.
+ * KinesisTestOptions} in order to run this if you want to test it with production setup. By default
+ * when no options are provided an instance of localstack is used.
  */
 @RunWith(JUnit4.class)
 public class KinesisIOIT implements Serializable {
-  private static int numberOfShards;
-  private static int numberOfRows;
-
   @Rule public TestPipeline pipelineWrite = TestPipeline.create();
   @Rule public TestPipeline pipelineRead = TestPipeline.create();
 
+  private static LocalStackContainer localstackContainer;
+
   private static KinesisTestOptions options;
-  private static final Instant now = Instant.now();
```

Comment:
Does this need to be `Instant.now()`? Might be preferable to just make it some constant to make it more repeatable.

Reply:
It's for the testing with real aws purpose. When a user wants to test Beam's KinesisIO with an existing stream then it would be best to read only the records that were written just in the time the test is running. If not, then it would be possible that write transform is broken and read works well (read some data from the past) and we have false-positive. It's still possible if a user uses a stream that receives records from elsewhere, but it cannot be helped.

Reply:
Ah I see. 

Is it necessary to support testing with an existing stream? It seems preferable to always create a stream for each test and clean it up after the test.

Reply:
Sorry I know this is scope-creep that doesn't directly relate to your change. We can keep it as-is for now, and file a jira to follow-up if it makes sense.

Reply:
Not a problem, it's already done. Thanks for your watchful eye, of course it's better to create a new stream for the test but I was stuck to the existing code. I also changed the InitialPositionInStream to TRIM_HORIZON, which reads the stream from the beginning instead of given timestamp.